packages/nextjs/src/utils/config.test.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright © 2024 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+import { enhanceConfig } from "./config"
+import { isProduction } from "./sdk"
+import { OryConfig } from "../types"
+
+jest.mock("./sdk", () => ({
+  isProduction: jest.fn(),
+}))
+
+describe("enhanceConfig", () => {
+  const originalEnv = process.env
+
+  beforeEach(() => {
+    jest.resetModules()
+    process.env = { ...originalEnv }
+  })
+
+  afterEach(() => {
+    process.env = originalEnv
+  })
+
+  it("should use forceSdkUrl if provided", () => {
+    const config: Partial<OryConfig> = {}
+    const result = enhanceConfig(config, "https://forced-url.com")
+    expect(result.sdk.url).toBe("https://forced-url.com")
+  })
+
+  it("should use NEXT_PUBLIC_ORY_SDK_URL if forceSdkUrl is not provided", () => {
+    process.env["NEXT_PUBLIC_ORY_SDK_URL"] = "https://public-sdk-url.com"
+    const config: Partial<OryConfig> = {}
+    const result = enhanceConfig(config)
+    expect(result.sdk.url).toBe("https://public-sdk-url.com")
+  })
+
+  it("should use ORY_SDK_URL if NEXT_PUBLIC_ORY_SDK_URL is not provided", () => {
+    process.env["ORY_SDK_URL"] = "https://sdk-url.com"
+    const config: Partial<OryConfig> = {}
+    const result = enhanceConfig(config)
+    expect(result.sdk.url).toBe("https://sdk-url.com")
+  })
+
+  it("should use __NEXT_PRIVATE_ORIGIN if not in production and forceSdkUrl is not provided", () => {
+    ;(isProduction as jest.Mock).mockReturnValue(false)
+    process.env["__NEXT_PRIVATE_ORIGIN"] = "https://private-origin.com/"
+    const config: Partial<OryConfig> = {}
+    const result = enhanceConfig(config)
+    expect(result.sdk.url).toBe("https://private-origin.com")
+  })
+
+  it("should use VERCEL_URL if __NEXT_PRIVATE_ORIGIN is not provided", () => {
+    ;(isProduction as jest.Mock).mockReturnValue(false)
+    process.env["VERCEL_URL"] = "vercel-url.com"
+    const config: Partial<OryConfig> = {}
+    const result = enhanceConfig(config)
+    expect(result.sdk.url).toBe("https://vercel-url.com")
+  })
+
+  xit("should use window.location.origin if VERCEL_URL is not provided", () => {
+    // Not sure if this works
```

Comment:
What do we need to check to make sure this works?

Reply:
Somehow be able to modify window.location.origin which I have not been able to get working in this test